### PR TITLE
Adding Deluxe Scarecrow to item effect ranges

### PR DESCRIPTION
**Pull request by alux35**
Moved scarecrow overlay calculation to the ModConfig file and added a calculation for the Deluxe Scarecrow. Changed code in ShowItemEffectRanges to pull the correct scarecrow for the current item and already placed items.

### CODE REVIEW

File: SDVModTest/ModConfig.cs [hunk 1]
```diff
@@ -85,5 +86,25 @@ class ModConfig
             new int[] { 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0 },
             new int[] { 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0 }
         };
+
+        public int[][] Scarecrow { get; set; } = CalculateScarecrow(17);
+
+        public int[][] DeluxeScarecrow { get; set; } = CalculateScarecrow(33);
+
+        private static int[][] CalculateScarecrow(int size)
```
Comment (cdaragorn) [inline SDVModTest/ModConfig.cs]: I love solving this with math! This function should be in the Tools class, please.

Comment (alux35) [reply SDVModTest/ModConfig.cs]: I didn't realize there was a tools class. I will definitely add it there. Thanks!

**Comment by jltaylor-us** [comment]
The deluxe scarecrow range calculation is incorrect, at least according to [some guy on reddit](https://www.reddit.com/r/StardewValley/comments/fakuq7/the_deluxe_scarecrow_range_seems_misunderstood/).

Here's an implementation using the distance calculation that he uses:

```c#
        private static int[][] GetCircularMask(int radius)
        {
            int size = 2 * radius + 1;
            int[][] result = new int[size][];
            for (int i = 0; i < size; i++)
            {
                result[i] = new int[size];
                for (int j = 0; j < size; j++)
                {
                    int val = (Math.Truncate(Math.Sqrt((radius - i) * (radius - i) + (radius - j) * (radius - j))) <= radius) ? 1 : 0;
                    result[i][j] = val;
                }
            }
            return result;
        }
```

Tangentially related to this PR: I'm not sure it's a great idea to have all of the ranges in ModConfig, leading to SMAPI generating a 726 line `config.json` file for the mod _before_ the scarecrow ranges were moved over there (and thousands of lines after), but whatever.  :shrug:

**Comment by Zamiell** [comment]
The Deluxe Scarecrow range is fixed now in Annosz's fork, so this PR is no longer necessary:
https://github.com/Annosz/UIInfoSuite2

**Pull request closed without merging**